packages/node_modules/glimmer-runtime/lib/syntax/builtins/each.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  CompileInto,
+  Statement as StatementSyntax
+} from '../../syntax';
+
+import * as Syntax from '../core';
+
+import {
+  LabelOpcode,
+  EnterOpcode,
+  PutArgsOpcode,
+  TestOpcode,
+  JumpUnlessOpcode,
+  JumpOpcode,
+  EvaluateOpcode,
+  ExitOpcode,
+  PushChildScopeOpcode,
+  PopScopeOpcode
+} from '../../compiled/opcodes/vm';
+
+import {
+  EnterListOpcode,
+  NextIterOpcode,
+  EnterWithKeyOpcode,
+  ExitListOpcode
+} from '../../compiled/opcodes/lists';
+
+import Environment from '../../environment';
+
+export default class EachSyntax extends StatementSyntax {
+  type = "each-statement";
+
+  public args: Syntax.Args;
+  public templates: Syntax.Templates;
+  public isStatic = false;
+
+  constructor({ args, templates }: { args: Syntax.Args, templates: Syntax.Templates }) {
+    super();
+    this.args = args;
+    this.templates = templates;
+  }
+
+  prettyPrint() {
+    return `#each ${this.args.prettyPrint()}`;
+  }
+
+  compile(compiler: CompileInto, env: Environment) {
+    //        PutArgs
+    //        EnterList(BEGIN, END)
+    // ITER:  Noop
+    //        NextIter(BREAK)
+    //        EnterWithKey(BEGIN, END)
+    // BEGIN: Noop
+    //        PushChildScope
+    //        Evaluate(default)
+    //        PopScope
+    // END:   Noop
+    //        Exit
+    //        Jump(ITER)
+    // BREAK: Noop
+    //        ExitList
+
+    let BEGIN = new LabelOpcode({ label: "BEGIN" });
+    let ITER = new LabelOpcode({ label: "ITER" });
+    let BREAK = new LabelOpcode({ label: "BREAK" });
+    let END = new LabelOpcode({ label: "END" });
+
+    compiler.append(new PutArgsOpcode({ args: this.args.compile(compiler, env) }));
+    compiler.append(new EnterListOpcode(BEGIN, END));
+    compiler.append(ITER);
+    compiler.append(new NextIterOpcode(BREAK));
+    compiler.append(new EnterWithKeyOpcode(BEGIN, END));
+    compiler.append(BEGIN);
+    compiler.append(new PushChildScopeOpcode());
+    compiler.append(new EvaluateOpcode({ debug: "default", block: this.templates.default }));
+    compiler.append(new PopScopeOpcode());
+    compiler.append(END);
+    compiler.append(new ExitOpcode());
+    compiler.append(new JumpOpcode({ target: ITER }));
+    compiler.append(BREAK);
+    compiler.append(new ExitListOpcode());
+  }
+}
```

Comment:
Hmm not sure if github bug or not but I just added the new line and the diff did not update.